### PR TITLE
add separate benchmark to compare the different implementations of binding

### PR DESCRIPTION
**Pull request by Munzey**
Basically shows that the binding function has same performance for both implementations (except of course if we use async in the coroutines version, which I added for reference).

### CODE REVIEW

**Comment by Munzey** [comment]
I'll come back to this once #29 is resolved

File: src/jvmBenchmark/kotlin/com/github/michaelbull/result/SuspendBindingBenchmark.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        private fun provideX(): Result<Int, Error> {
+            Thread.sleep(time)
+            return Ok(1)
+        }
+        private fun provideY(): Result<Int, Error> {
+            Thread.sleep(time)
+            return Ok(2)
+        }
+        private suspend fun suspendedProvideX(): Result<Int, Error> {
```
Comment (Munzey) [inline src/jvmBenchmark/kotlin/com/github/michaelbull/result/SuspendBindingBenchmark.kt:67]: I recall now that Roman recommends this should be named `provideX()` and the blocking version `provideXBlocking()`

**Comment by michaelbull** [comment]
Hi @Munzey, are there any plans to return to this PR or can we close it for now?

**Comment by Munzey** [comment]
hey @michaelbull yes was planning to come back to this, its just blocked by #38 which there is a pr up now #41.
Obv #41 hasn't gotten much attention lately so will see if I can get that moving - otherwise yeah maybe we look to close this out end of the week?

**Comment by michaelbull** [comment]
Merged in e1fd6387e58fcdb24d09dbbde4bda47ada9c850c

Preview of benchmarks:

image

**Pull request closed without merging**